torch/csrc/api/include/torch/nn/options/conv.h

```diff
@@ -11,27 +12,31 @@ namespace nn {
 /// Options for a `D`-dimensional convolution module.
 template <size_t D>
 struct ConvOptions {
```

Comment:
Sometimes we have TORCH_API before classname, sometimes not...

Reply:
Right this is actually intentional - in https://github.com/pytorch/pytorch/blob/master/CONTRIBUTING.md#windows-development-tips: 
> However, there is one important counterexample to this principle: if you want a templated function to be instantiated at the call site, do NOT mark it with *_API (if you do mark it, you'll have to explicitly instantiate all of the specializations used by the call sites.)

Although we do explicitly instantiate all the specializations of `ConvOptions` in `torch/csrc/api/src/nn/options/conv.cpp`, I think it might still be a good idea to not put TORCH_API here, which is also consistent with how other templatized Options are implemented.